websites/K/KaryaKarsa/presence.ts

```diff
@@ -0,0 +1,231 @@
+        paths: [
+          {
+            name: "",
+            state: "My Vouchers"
+          },
+          {
+            name: "add",
+            state: "Making new Voucher..."
+          },
+          {
+            name: "*",
+            state: "Editing my Voucher..."
+          }
+        ]
+      },
+      {
+        name: "overlay",
+        state: "Stream Overlay"
+      },
+      {
+        name: "supports",
+        state: "Supports History"
+      },
+      {
+        name: "bank",
+        state: "Payment"
+      },
+      {
+        name: "profile",
+        state: "Editing Profile"
+      },
+      {
+        name: "settings",
+        state: "Settings"
+      }
+    ]
+  },
+  getUserHeader = ".q-page-container.q-pb-xl.main",
+  getUserHeaderHome =
+    ".title.q-toolbar__title.ellipsis a.no-underline.text-primary.router-link-active",
+  // This function act like switch function but for objects Array (yea its a lil bit complicated)
+  stateSetter = (
+    paths: string[],
+    arrayPath: Array<{
+      name: string;
+      state?: string;
+      paths?: Array<{
+        name?: string;
+        state?: string;
+      }>;
+    }>
+  ): string => {
+    const result = arrayPath.find((p) => p.name === (paths[1] || ""));
+    if (result.state) return result.state;
+    return result.paths.find((p) => {
+      const b: boolean = p.name === (paths[2] || "");
+      if (!b && !!paths[2]) return p.name === "*";
+      return b;
+    }).state;
+  };
+
+presence.on("UpdateData", () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "main",
+    startTimestamp: browsingTimestamp
+  },
+  paths: string[] = window.location.pathname.split("/").splice(1);
+
+  switch (paths[0].toLowerCase()) {
+    case "":
+      presenceData.details = "Looking at HomePage";
+      break;
+    case "feed":
+      presenceData.details = "Looking at Feeds";
+      break;
+    case "profile":
+      presenceData.details = "Looking at Profile";
+      break;
+    case "about":
+    case "terms":
+      presenceData.details = "Looking at Terms of Service";
+      break;
+    case "dashboard":
+      presenceData.details = "Looking at Dashboard";
+      presenceData.state = stateSetter(paths, pathData.dashboard);
+      break;
+    case "discover": {
+      presenceData.details = "Looking at Discover / Search";
+      const searchBar = <HTMLInputElement>(
+        document.querySelector(
+          ".q-field.q-input.q-field--filled.q-field--square.q-field--float.q-validation-component input.q-field__native.q-placeholder"
+        )
+      );
+      if (searchBar?.value)
+        presenceData.state = `Searching for "${searchBar.value}"`;
+      break;
+    }
+    case "user":
+      presenceData.details = "Looking at My Detail";
+      presenceData.state = stateSetter(paths, pathData.user);
+      break;
+    default: {
+      // Default (If the path isnt defined here. Will be replaced with this!)
+      presenceData.details = "Browsing Pages";
```

Comment:
If you wanna do what the comment above says (replace if not set), you should be doing this
```suggestion
      presenceData.details ??= "Browsing Pages";
```